x/incentive/migrations/v2/store.go

```diff
@@ -0,0 +1,142 @@
+package v2
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/kava-labs/kava/x/incentive/types"
+)
+
+// MigrateStore performs in-place migrations from incentive ConsensusVersion 1 to 2.
+func MigrateStore(ctx sdk.Context, storeKey storetypes.StoreKey, cdc codec.BinaryCodec) error {
+	store := ctx.KVStore(storeKey)
+
+	// Migrate earn claims
+	if err := MigrateEarnClaims(store, cdc); err != nil {
+		return err
+	}
+
+	// Migrate accrual times
+	if err := MigrateAccrualTimes(store, cdc, types.CLAIM_TYPE_EARN); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// MigrateEarnClaims migrates earn claims from v1 to v2
+func MigrateEarnClaims(store sdk.KVStore, cdc codec.BinaryCodec) error {
+	newStore := prefix.NewStore(store, types.GetClaimKeyPrefix(types.CLAIM_TYPE_EARN))
+
+	oldStore := prefix.NewStore(store, EarnClaimKeyPrefix)
+	iterator := sdk.KVStorePrefixIterator(oldStore, []byte{})
+	defer iterator.Close()
+	for ; iterator.Valid(); iterator.Next() {
+		var c types.EarnClaim
+		cdc.MustUnmarshal(iterator.Value(), &c)
+
+		if err := c.Validate(); err != nil {
+			return fmt.Errorf("invalid v1 EarnClaim: %w", err)
+		}
```

Comment:
Nice validations. I see what you mean about the tests not being very reassuring. Given mainnet state could be buggy this kind of defensive assumption checking is probably the way to go.